modules/change_detection/src/dynamic_change_detector.js

```
@@ -0,0 +1,245 @@
+import {isPresent, isBlank, BaseException, FunctionWrapper} from 'facade/lang';
+import {List, ListWrapper, MapWrapper, StringMapWrapper} from 'facade/collection';
+import {ContextWithVariableBindings} from './parser/context_with_variable_bindings';
+
+import {ArrayChanges} from './array_changes';
+import {KeyValueChanges} from './keyvalue_changes';
+
+import {
+  ProtoRecord,
+  RECORD_TYPE_SELF,
+  RECORD_TYPE_PROPERTY,
+  RECORD_TYPE_INVOKE_METHOD,
+  RECORD_TYPE_CONST,
+  RECORD_TYPE_INVOKE_CLOSURE,
+  RECORD_TYPE_INVOKE_PURE_FUNCTION,
+  RECORD_TYPE_INVOKE_FORMATTER,
+  RECORD_TYPE_STRUCTURAL_CHECK,
+  ProtoChangeDetector
+  } from './proto_change_detector';
+
+import {ChangeDetector, ChangeRecord, ChangeDispatcher} from './interfaces';
+import {ExpressionChangedAfterItHasBeenChecked, ChangeDetectionError} from './exceptions';
+
+class SimpleChange {
+  previousValue:any;
+  currentValue:any;
+
+  constructor(previousValue:any, currentValue:any) {
+    this.previousValue = previousValue;
+    this.currentValue = currentValue;
+  }
+}
+
+export class DynamicChangeDetector extends ChangeDetector {
+  dispatcher:any;
+  formatters:Map;
+  children:List;
+  values:List;
+  protos:List<ProtoRecord>;
+  parent:ChangeDetector;
+
+  constructor(dispatcher:any, formatters:Map, protoRecords:List<ProtoRecord>) {
+    this.dispatcher = dispatcher;
+    this.formatters = formatters;
+    this.values = ListWrapper.createFixedSize(protoRecords.length + 1);
+    this.protos = protoRecords;
+
+    this.children = [];
+  }
+
+  addChild(cd:ChangeDetector) {
+    ListWrapper.push(this.children, cd);
+    cd.parent = this;
+  }
+
+  removeChild(cd:ChangeDetector) {
+    ListWrapper.remove(this.children, cd);
+  }
+
+  remove() {
+    this.parent.removeChild(this);
+  }
+
+  setContext(context:any) {
+    this.values[0] = context;
+  }
+
+  detectChanges() {
```

Per our discussion, pass in a boolean, representing if the changes should be delivered to dispatcher or should throw an error.